client/src/app/containers/MapViewContainer/MapViewContainer.js

```diff
@@ -0,0 +1,200 @@
+// get API data from police
+import React, {useEffect, useState} from "react";
+import {GoogleMap,CrimeList, CrimeSearchBar} from "../../components";
+import PoliceAPI from "../../services/PoliceAPI";
+import GeocodeAPI from "../../services/GeocodeAPI";
+import localAPI from "../../services/localAPI";
+import { CrimeUpdateForm } from "../../components/crimeUpdateForm/crimeUpdatePage/crimeUpdatePage";
+import { CrimeBarChart } from "../../components/crimeBarChart/crimeBarChart";
+import crimeLocationObj from "./crimeObj";
+const uuidv4 = require('uuid/v4');
+
+export const MapContext = React.createContext(null);
+
+//"other-theft" "theft-from-the-person"
+
+
+export const  MapView = ({match}) => {
+// set search as the default mode
+const initialMode = match.params.add ? match.params.add : 'search';
+const [crimeLocations, setCrimeLocations] = useState([]);
+const [mapMode, setMapMode] = useState(initialMode);
+const [crimeCentre, setCrimeCentre] = useState({lat: 52.397, lng: 0.4196, rad:0.005});  
+
+const setNewMode = (mapMode) => {
+  // if moving to add or map search clear crime locations 
+  if (mapMode === 'add'|| mapMode ==='search') setCrimeLocations([]);
+  setMapMode(mapMode);
+}
+
+const resetAddMode = () => {
+  setMapMode('add');
+}
+
+const resetSearchMode = () => {
+  setMapMode('search');
+}
+
+// reduce crimes by location
+useEffect(()=> {
+  // get Police data based on Polygon coordinates
+  if (mapMode === 'search' | mapMode === 'statistics') {
+    PoliceAPI.getCrimes(crimeCentre)
+    .then(crimes => {
+      const crimeCoordinates  = crimeLocationObj.parsePoliceCrimes(crimes);
+      setCrimeLocations(crimeLocations =>[...crimeLocations,...crimeCoordinates]);
+    })
+    localAPI.getCrimes(crimeCentre).then((crimes)=> {
```

Comment:
I would centralize all crimes handling in the `localApi`. For example, from the client you only do a single request to the `localAPI` and it does the fetch to the Police one, appending the local crimes after.